File: src/main/java/com/vjiki/music/dto/MessageReactionResponse.java
package com.vjiki.music.dto;

import java.time.OffsetDateTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MessageReactionResponse {
    private UUID messageId;
    private UUID userId;
    private String emoji;

    @JsonProperty("created_at")
    private OffsetDateTime createdAt;
}

File: src/main/kotlin/com/vjiki/music/dto/MessageReactionResponse.kt
package com.vjiki.music.dto

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.OffsetDateTime
import java.util.UUID

data class MessageReactionResponse(
    val messageId: UUID,
    val userId: UUID,
    val emoji: String,
    @JsonProperty("created_at")
    val createdAt: OffsetDateTime?
)
